pycytominer/cyto_utils/load.py

```diff
@@ -1,10 +1,54 @@
 import csv
 import gzip
+import pathlib
 import numpy as np
 import pandas as pd
+from typing import Union
 
+def is_path_a_parquet_file(file: Union[str, pathlib.Path]) -> bool:
+    """Checks if the provided file path is a parquet file.
 
-def infer_delim(file):
+    Identify parquet files by inspecting the file extensions.
+    If the file does not end with `parquet`, this will return False, else True.
+
+    Parameters
+    ----------
+    file : Union[str, pathlib.Path]
+        path to parquet file
+
+    Returns
+    -------
+    bool
+        Returns True if the file path contains `.parquet`, else it will return
+        False
+
+    Raises
+    ------
+    TypeError
+        Raised if a non str object is passed in the `file` parameter
+    FileNotFoundError
+        Raised if the provided path in the `file` does not exist
+    """
+    # type checking
+    accepted_type = (str, pathlib.Path)
+    if not isinstance(file, accepted_type):
```

Comment:
Consider using the tuple directly within `isinstance` to avoid creating a variable which is only used once.

```suggestion
    if not isinstance(file, (str, pathlib.Path)):
```